src/plugin/ethTransaction.ts

```diff
@@ -392,8 +429,8 @@ export class EthereumTransaction implements Interfaces.Transaction {
       }
       const signature = toEthereumSignatureNative({
         v: bufferToInt(v),
-        r,
-        s,
+        r: bufferPadStart(r, 32), // The validate signature logic expects 32 bytes. Some r and s values can result in 31 bytes - pad the number to ensure it's seen as valid.
```

Comment:
This code is replicated later and also belongs at a higher level.
We want to protect against others falling into the same trap too.
It looks like this adjustment should be added to toEthereumSignatureNative() right before ...
  if (isValidEthereumSignatureNative(signature)) {

Reply:
Note: The whole point of the helper functions like toEthereumSignatureNative() is to handle this stuff

Reply:
toEthereumSignatureNative takes a fully structured signature. This fix needs to be applied to the r and s values only. To implement your suggestion I'd need to either deconstruct the signature again in toEthereumSignatureNative or take a new input format. My intention is to come back and fix this all properly. The method by which we're validating signatures is poorly constructed to begin with. Almost all libraries take what was signed along with the signature so that the signature can be checked in context of what was signed - what we have at the moment is a bit of a hack job imo. Since this is not the primary focus of this pull request I chose to adddress the current problem as quickly as possible and not go down an entirly new rabit hole.